src/model/src/SolidShapes.cpp

```diff
@@ -194,6 +197,91 @@ namespace iDynTree
 
     const std::string& ExternalMesh::getFilename() const { return filename; }
 
+    std::string ExternalMesh::getFileLocationOnLocalFileSystem() const
+    {
+        bool isWindows = false;
+    #ifdef _WIN32
+        isWindows = true;
+    #endif
+
+        std::unordered_set<std::string> pathList;
+        std::vector<std::string> envList = {"GAZEBO_MODEL_PATH", "ROS_PACKAGE_PATH", "AMENT_PREFIX_PATH"};
+
+        for (size_t i = 0; i < envList.size(); ++i)
+        {
+            const char * env_var_value = std::getenv(envList[i].c_str());
+
+            if (env_var_value)
+            {
+                std::stringstream env_var_string(env_var_value);
+
+                std::string individualPath;
+
+                while(std::getline(env_var_string, individualPath, isWindows ? ';' : ':'))
+                {
+                    pathList.insert(individualPath);
+                }
+            }
+        }
+
+
+        auto cleanPathSeparator = [isWindows](const std::string& filename)->std::string
+        {
+            std::string output = filename;
+            char pathSeparator = isWindows ? '\\' : '/';
+            char wrongPathSeparator = isWindows ? '/' : '\\';
+
+            for (size_t i = 0; i < output.size(); ++i)
+            {
+                if (output[i] == wrongPathSeparator)
+                {
+                    output[i] = pathSeparator;
+                }
+            }
+
+            return filename;
+
+        };
+
+        auto isFileExisting = [cleanPathSeparator](const std::string& filename)->bool
+        {
+            if (FILE *file = fopen(cleanPathSeparator(filename).c_str(), "r")) {
+                fclose(file);
+                return true;
+            } else {
+                return false;
+            }
+        };
+
+        auto getFilePath = [isFileExisting](const std::string& filename, const std::string& prefixToRemove, const std::unordered_set<std::string>& paths)
+        {
+            if(isFileExisting(filename))
+            {
+                return filename;
+            }
+
```

Comment:
This ensure that the current use case of specifying files without a `package://`  URI file is still supported, right?

Reply:
Yes, if ``filename`` is already the path of a file, it simply returns it as it is.